internal/source/image.go

```diff
@@ -164,22 +190,53 @@ func (i *Image) getBundleImageDigest(pod *corev1.Pod) (string, error) {
 	return "", fmt.Errorf("bundle image digest not found")
 }
 
-func (i *Image) getPodLogs(ctx context.Context, pod *corev1.Pod) ([]byte, error) {
-	logReader, err := i.KubeClient.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{}).Stream(ctx)
+func (i *Image) getPod(ctx context.Context, job *batchv1.Job) (*corev1.Pod, error) {
+	podList := corev1.PodList{}
+	ls, err := metav1.LabelSelectorAsSelector(job.Spec.Selector)
 	if err != nil {
-		return nil, fmt.Errorf("get pod logs: %v", err)
+		return nil, err
 	}
-	defer logReader.Close()
-	buf := &bytes.Buffer{}
-	if _, err := io.Copy(buf, logReader); err != nil {
+	if err := i.Client.List(ctx, &podList, client.InNamespace(job.Namespace), client.MatchingLabelsSelector{Selector: ls}); err != nil {
 		return nil, err
 	}
-	return buf.Bytes(), nil
+	if len(podList.Items) == 0 {
+		return nil, nil
+	}
```

Comment:
Could you help me understand why no pods found for a specific job is not an error?

Reply:
I need to double check if we need to also handle the case where the job completed, the pod completed, and then the pod was deleted without the job being deleted.

Reply:
And... we do. Good catch!